### ISSUE TITLE
expected one argument

### ISSUE THREAD

**Issue by ahworld22** [ZelongZeng/PLCD#4]
hello ,what is this problem and how to deal with？
![Uploading image.png…]()

**Comment by ZelongZeng** [comment]
Hi, you need to input a argument to `--name`.

**Comment by ahworld22** [comment]
OK，if i use windows，it also can be done right？



---- Replied Message ----
| From | Zelong ***@***.***> |
| Date | 09/20/2024 10:02 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [ZelongZeng/PLCD] expected one argument (Issue #4) |

Hi, you need to input a argument to --name.

**Comment by ZelongZeng** [comment]
You only need to enter it in the command line.
Please refer to the corresponding section in the "README" for the specific command.

**Comment by ahworld22** [comment]
> 您只需在命令行中输入它。具体命令请参考 “README” 中的相应部分。
I donn't know what should I input with --name

**Comment by ahworld22** [comment]
I download your models and put it into file,then I input the name of G2D and it runs!But it says :
CUDA out of memory. Tried to allocate 28.00 MiB (GPU 0; 8.00 GiB total capacity; 12.49 GiB already allocated; 0 bytes f
ree; 12.55 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
I wanna know if i must use apex?And if I use apex must i install it with ext-cudnn?

**Comment by ZelongZeng** [comment]
You want to train the model, right?
If you want to use apex, you can input `--fp16`, it will saves around 50% memory but reduce the result. 
Another approach is to use multiple GPUs for training. You can search for the commented “nn.DataParallel” code in the relevant .py files, uncomment it, and then use the `--gpu_ids` command-line argument to select the desired GPUs. 
Of course, I would highly recommend that you modify the code to use PyTorch DDP for multi-GPU training (I haven’t implemented PyTorch DDP in this project).

**Comment by ahworld22** [comment]
OK thanks a lot，i will try it.If I use apex must i install apex with install --cuda_ext --cpp_ext？or i only need to install apex itself?



---- Replied Message ----
| From | Zelong ***@***.***> |
| Date | 09/21/2024 05:38 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [ZelongZeng/PLCD] expected one argument (Issue #4) |

You want to train the model, right?
If you want to use apex, you can input --fp16, it will saves around 50% memory but reduce the result.
Another approach is to use multiple GPUs for training. You can search for the commented “nn.DataParallel” code in the relevant .py files and then use the --gpu_ids command-line argument to select the desired GPUs.
Of course, I would highly recommend you for modifying the code to use PyTorch DDP for multi-GPU training.

**Comment by ZelongZeng** [comment]
Hi, normally, these two options are selected during the installation of Apex, but I’m not sure if they are mandatory requirements. Please refer to the official Apex documentation for details.

**Comment by ahworld22** [comment]
OK，because i'm using windows it seems difficult to install apex with that tow options.I think maybe i will try PythonDDP.Thank you!



---- Replied Message ----
| From | Zelong ***@***.***> |
| Date | 09/21/2024 21:07 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [ZelongZeng/PLCD] expected one argument (Issue #4) |

Hi, normally, these two options are selected during the installation of Apex, but I’m not sure if they are mandatory requirements. Please refer to the official Apex documentation for details.

**Comment by ahworld22** [comment]
Hello!If i want to use ddp ,how can i modify the code?













At 2024-09-21 05:38:19, "Zelong Zeng" ***@***.***> wrote:

You want to train the model, right?
If you want to use apex, you can input --fp16, it will saves around 50% memory but reduce the result.
Another approach is to use multiple GPUs for training. You can search for the commented “nn.DataParallel” code in the relevant .py files and then use the --gpu_ids command-line argument to select the desired GPUs.
Of course, I would highly recommend you for modifying the code to use PyTorch DDP for multi-GPU training.

**Comment by ahworld22** [comment]
hello，i have the question that whats the different of train step 1，2 and 3？Also，what should I put to the old_name and name?




| |
***@***.***
|
|
***@***.***
|

**Comment by ZelongZeng** [comment]
Hi, in step-2 and step-3, we apply peer-learning (distillation) by using the pre-trained model from the last step. Please refer to Section **III. OUR METHOD** of our paper for specific details.

**Comment by ahworld22** [comment]
Yes, I used the model with the highest accuracy from the previous step as the teacher for training. I trained the second step on the model that achieved an accuracy of 90% in the first step, but the final accuracy was very low. I don't know what caused this. However, I noticed that I didn't do the prepare_limited_view step, could it be the problem?













At 2024-10-07 15:52:37, "Zelong Zeng" ***@***.***> wrote:

Hi, in step-2 and step-3, we apply peer-learning (distillation) by using the pre-trained model from the last step. Please refer to Section III. OUR METHOD of our paper for specific details.
